src/movies.js

```diff
@@ -1,23 +1,75 @@
 // Iteration 1: All directors? - Get the array of all directors.
-// _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+// _Bonus_: It seems some of the directors had directed multiple movies so they will 
+// pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    const movie = moviesArray.map((movie) => movie.director)
+    return movie
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    const moviesStevenSpielberg = [];
+     if (moviesArray.length === 0) {
```

Comment:
The `if (moviesArray.length === 0) { return 0;}` block is redundant in this function. The filter function will not execute if the moviesArray is empty, so you do not need to check whether the array is empty or not. The function will work fine without this block.